Navigation: Update focus style, Get WordPress button mobile style

ryelle opened the pull request:
Fixes #372, fixes #363. This updates the focus style across the header, and removes the special styling on the mobile "Get WordPress" button.

I'm using the "current item" color for the focus state, which is `--wp--preset--color--blueberry-2` on the black background, `--wp--preset--color--deep-blueberry` on the white background, and `--wp--preset--color--white` on the blue background.

| | Default | Blue bg | White bg |
|---|---|---|---|
| Top level item focus | ![branch-default-1-large-menu-focus](https://user-images.githubusercontent.com/541093/228328732-73be22d4-48f2-424d-a138-e2f1432e3faf.png) | ![branch-blue-1-large-menu-focus](https://user-images.githubusercontent.com/541093/228328710-dee38bd6-1faf-4947-9fef-97ee00c0b386.png) | ![branch-white-1-large-menu-focus](https://user-images.githubusercontent.com/541093/228328748-3802f0fb-3add-4969-9870-751a3b606a38.png) |
| Menu open, submenu item focus | ![branch-default-2-large-menu-open](https://user-images.githubusercontent.com/541093/228328734-71805791-3440-4989-859f-429fc007c60a.png) | ![branch-blue-2-large-menu-open](https://user-images.githubusercontent.com/541093/228328713-a02ff21b-650a-46f9-a33c-82153bbdfe34.png) | ![branch-white-2-large-menu-open](https://user-images.githubusercontent.com/541093/228328753-f8fb333f-1027-4884-8f96-01688695adb3.png) |
| Get WordPress focus | ![branch-default-3-large-button-focus](https://user-images.githubusercontent.com/541093/228328736-f1de8ec6-0a77-46e7-92dc-a92fe1ff1339.png) | ![branch-blue-3-large-button-focus](https://user-images.githubusercontent.com/541093/228328715-c156e87b-5d18-4b94-a8df-ba0914733b1b.png) | ![branch-white-3-large-button-focus](https://user-images.githubusercontent.com/541093/228328756-e0a7248d-07b8-4801-96cd-26abb7e2569b.png) |
| Small screen, logo focus | ![branch-default-4-small-logo-focus](https://user-images.githubusercontent.com/541093/228328738-98644e9b-7a28-4d70-9bed-6cf043472ebd.png) | ![branch-blue-4-small-logo-focus](https://user-images.githubusercontent.com/541093/228328717-87b392ac-3935-4a01-9ee6-ca2add1b1d4e.png) | ![branch-white-4-small-logo-focus](https://user-images.githubusercontent.com/541093/228328760-7101af5a-bcc8-44cc-a0d5-d9284118f319.png) |
| Small screen, menu open, top level item focus | ![branch-default-5-small-menu-top-level](https://user-images.githubusercontent.com/541093/228328740-89d7f10e-177e-4463-b1e8-38528d6e1ba4.png) | ![branch-blue-5-small-menu-top-level](https://user-images.githubusercontent.com/541093/228328721-8f31203c-d3ed-4bab-bf79-130ae6e90a82.png) | ![branch-white-5-small-menu-top-level](https://user-images.githubusercontent.com/541093/228328761-a6c027ef-b550-4e3f-a3f3-ee17c3d61949.png) |
| Small screen, menu open, submenu item focus | ![branch-default-6-small-menu-2nd-level](https://user-images.githubusercontent.com/541093/228328743-6d06f421-3a54-47c1-9dcb-771f7f3e443e.png) | ![branch-blue-6-small-menu-2nd-level](https://user-images.githubusercontent.com/541093/228328728-8899f1f6-e1c8-455b-a986-449acb80f7b9.png) | ![branch-white-6-small-menu-2nd-level](https://user-images.githubusercontent.com/541093/228328762-1d7c47f3-24b0-4442-acd9-998cc2655e2e.png) |

The spacing around the small screen menu items have also been tweaked, but I've kept the highlight at the edge of the screen

<img width="200" src="https://user-images.githubusercontent.com/541093/228360030-dd831f28-423d-43bd-ae7e-2f4a8446abda.png" />

ryelle committed:
Navigation: Update focus style

ryelle committed:
Get WordPress: Keep consistent styles between large and small screens

StevenDufresne commented:
I don't think i see another issue for this, but can we address the fact that the focus indicator is cut off and not aligned on the bottom (off by a couple pixels?):

<img width="191" alt="Screenshot 2023-03-29 at 10 31 26 AM" src="https://user-images.githubusercontent.com/1657336/228403448-aa22857a-2a49-4bb0-889d-368aa288281e.png">

Seems to be related to a border bottom on the `li`.

I would also prefer if the outline was closer to the text, but that could probably be done later.

jasmussen commented:
Thanks for the PR, and excellent catch Steve! The box shadow needs to be inset, that also corrects the border radius. So instead of:

> box-shadow: 0 0 0 1.5px var(--wp-global-header--link-color--active);

it should be 

> box-shadow: inset 0 0 0 1.5px var(--wp-global-header--link-color--active);

That should fix the crop.

ryelle committed:
Inset the focus border to prevent cropping

The exception to this is the button, which would loose accessible contrast if the border was inset.

ryelle commented:
I've switched the box-shadow to inset, so it's not cut off anymore:

<img width="173" alt="Screenshot 2023-03-29 at 4 54 41 PM" src="https://user-images.githubusercontent.com/541093/228665117-a225ad38-231f-4766-b680-19f0d0f9bc15.png">

Will merge this now to keep things moving, but if we need to iterate more we can.
